rust/lance-core/src/utils/blob.rs

```diff
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: Copyright The Lance Authors
+
+use object_store::path::Path;
+
+/// Format a dedicated blob sidecar path for a data file.
+///
+/// Layout: `<base>/<data_file_key>/<blob_id>.raw`
+/// - `base` is typically the dataset's data directory.
+/// - `data_file_key` is the stem of the data file (without extension).
```

Comment:
Interesting.  I hadn't expected path to include `data_file_key` but I don't see any harm in it and it could maybe be useful for things like cleanup?